File: tests/testfiles/complex3.py
def factorial(n : int) -> int:
	if not n:
		return 1
	m = n - 1
	return factorial(m)

factorial(5)
x : int = factorial(10)

print(x, factorial(7))

i = 0
while i < 100:
	i : int
	if i % 7 == 0:
		z : float = factorial(i)
		z = z * z
	else:
		z : float = factorial(i)
		z = z * z * z
	
x = 2

print()

File: tests/testfiles/complex3.cpp
#include <iostream>

int factorial(int n)
{
    if(! n)
    {
        return 1;
    }
    m = n - 1;
    return factorial(m);
}

int main()
{
    int x;
    int i;
    float z;
    float z;
    factorial(5);
    x = factorial(10);
    std::cout << x << factorial(7) << std::endl;
    i = 0;
    while(i < 100)
    {
        if(i % 7 == 0)
        {
            z = factorial(i);
            z = z * z;
        }
        else
        {
            z = factorial(i);
            z = z * z * z;
        }
    }
    x = 2;
    std::cout << std::endl;
    return 0;
}
